Security/LoginManager.php

```diff
@@ -27,16 +28,23 @@
  */
 class LoginManager implements LoginManagerInterface
 {
-    private $securityContext;
+    /**
+     * @var SecurityContextInterface|TokenStorageInterface
+     */
+    private $tokenStorage;
     private $userChecker;
     private $sessionStrategy;
     private $container;
 
-    public function __construct(SecurityContextInterface $context, UserCheckerInterface $userChecker,
+    public function __construct($tokenStorage, UserCheckerInterface $userChecker,
                                 SessionAuthenticationStrategyInterface $sessionStrategy,
                                 ContainerInterface $container)
     {
-        $this->securityContext = $context;
+        if (!$tokenStorage instanceof TokenStorageInterface && !$tokenStorage instanceof SecurityContextInterface) {
+            throw new \InvalidArgumentException(sprintf('Argument 1 of %s should be an instance of TokenStorageInterface or SecurityContextInterface', __CLASS__));
```

Comment:
FQCN should be used (and IMO, there is no need to mention ``__CLASS__``. the error location already shows you where the error is